app/models/katello/glue/candlepin/pool.rb

```diff
@@ -187,8 +187,23 @@ def import_hosts
       end
 
       def import_managed_associations
+        start = Time.now
         import_hosts
+        Rails.logger.info("Imported hosts in #{Time.now - start} seconds")
+
+        start = Time.now
         create_activation_key_associations
+        Rails.logger.info("Imported activation keys in #{Time.now - start} seconds")
+      end
+
+      def delete_associations
+        batch_size = 10_000
+        facet_pool_count = subscription_facet_pools.count
+        iterations = (facet_pool_count.to_f / batch_size).ceil
+
+        (1..iterations).each do
+          subscription_facet_pools.limit(batch_size).delete_all
```

Comment:
did you see a big performance difference when deleting in batches versus deleting all?

Reply:
I didn't compare but thought it'd be better to have smaller transactions. Let me actually test that out

Reply:
So I had 50K SubscriptionFacetPools on my pool, and the batch delete was actually twice as slow!

delete_all
```
2020-06-11T13:07:23 [D|sql|]   Katello::SubscriptionFacetPool Destroy (29.1ms)  DELETE FROM "katello_subscription_facet_pools" WHERE "katello_subscription_facet_pools"."pool_id" = $1  [["pool_id", 394]]  
```

batched:

```
2020-06-11T13:11:48 [D|sql|]   Katello::SubscriptionFacetPool Destroy (18.2ms)  DELETE FROM "katello_subscription_facet_pools" WHERE "katello_subscription_facet_pools"."id" IN (SELECT "katello_subscription_facet_pools"."id" FROM "katello_subscription_facet_pools" WHERE "katello_subscription_facet_pools"."pool_id" = $1 LIMIT $2)  [["pool_id", 401], ["LIMIT", 10000]]
2020-06-11T13:11:48 [D|sql|]   Katello::SubscriptionFacetPool Destroy (17.3ms)  DELETE FROM "katello_subscription_facet_pools" WHERE "katello_subscription_facet_pools"."id" IN (SELECT "katello_subscription_facet_pools"."id" FROM "katello_subscription_facet_pools" WHERE "katello_subscription_facet_pools"."pool_id" = $1 LIMIT $2)  [["pool_id", 401], ["LIMIT", 10000]]
2020-06-11T13:11:48 [D|sql|]   Katello::SubscriptionFacetPool Destroy (15.3ms)  DELETE FROM "katello_subscription_facet_pools" WHERE "katello_subscription_facet_pools"."id" IN (SELECT "katello_subscription_facet_pools"."id" FROM "katello_subscription_facet_pools" WHERE "katello_subscription_facet_pools"."pool_id" = $1 LIMIT $2)  [["pool_id", 401], ["LIMIT", 10000]]
2020-06-11T13:11:48 [D|sql|]   Katello::SubscriptionFacetPool Destroy (14.5ms)  DELETE FROM "katello_subscription_facet_pools" WHERE "katello_subscription_facet_pools"."id" IN (SELECT "katello_subscription_facet_pools"."id" FROM "katello_subscription_facet_pools" WHERE "katello_subscription_facet_pools"."pool_id" = $1 LIMIT $2)  [["pool_id", 401], ["LIMIT", 10000]]
2020-06-11T13:11:48 [D|sql|]   Katello::SubscriptionFacetPool Destroy (12.3ms)  DELETE FROM "katello_subscription_facet_pools" WHERE "katello_subscription_facet_pools"."id" IN (SELECT "katello_subscription_facet_pools"."id" FROM "katello_subscription_facet_pools" WHERE "katello_subscription_facet_pools"."pool_id" = $1 LIMIT $2)  [["pool_id", 401], ["LIMIT", 10000]]
```

I guess delete_all is all we need

Reply:
interesting!